supervisor/poller.py

```diff
@@ -78,15 +81,28 @@ def initialize(self):
         self._poller = select.poll()
         self.READ = select.POLLIN | select.POLLPRI | select.POLLHUP
         self.WRITE = select.POLLOUT
+        self.readables = set()
+        self.writables = set()
 
     def register_readable(self, fd):
         self._poller.register(fd, self.READ)
+        self.readables.add(fd)
 
     def register_writable(self, fd):
         self._poller.register(fd, self.WRITE)
+        self.writables.add(fd)
 
-    def unregister(self, fd):
+    def unregister_readable(self, fd):
+        self.readables.discard(fd)
         self._poller.unregister(fd)
+        if fd in self.writables:
+            self._poller.register(fd, self.WRITE)
```

Comment:
If I am reading this correctly, it seems like unregistering a readable causes all writables to be (re)registered(?).  Could you please explain this?  Same thing in `unregister_writable` below.